corehq/apps/repeaters/tasks.py

```diff
@@ -60,9 +61,10 @@ def process_repeat_record(repeat_record):
         if repeat_record.repeater.doc_type.endswith(DELETED_SUFFIX):
             if not repeat_record.doc_type.endswith(DELETED_SUFFIX):
                 repeat_record.doc_type += DELETED_SUFFIX
+                repeat_record.save()
         else:
-            repeat_record.fire()
-        repeat_record.save()
+            if repeat_record.state == RECORD_PENDING_STATE or repeat_record.state == RECORD_FAILURE_STATE:
```

Comment:
@benrudolph notifying you if you have context on this. I saw that you made some changes to this last. More context [here](https://manage.dimagi.com/default.asp?248743#bugevent_1297570)